File: C#/1302_DeepestLeavesSum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int DeepestLeavesSum(TreeNode root) {
        Queue<(int, TreeNode)> found = new Queue<(int, TreeNode)>();
        Dictionary<int,int> sumByLevel = new Dictionary<int,int>();
        int dl = 1;
        found.Enqueue((1,root));
        while(found.Count > 0){
            (int, TreeNode) I = found.Dequeue();
            if(sumByLevel.ContainsKey(I.Item1)) {
                sumByLevel[I.Item1] += I.Item2.val;
            }
            else {
                dl = I.Item1;
                sumByLevel.Add(I.Item1, I.Item2.val);
            }
            if (I.Item2.left != null)
                found.Enqueue((I.Item1 + 1, I.Item2.left));
            if (I.Item2.right != null)
                found.Enqueue((I.Item1 + 1, I.Item2.right));
        }
        return sumByLevel[dl];
    }
}

File: JavaScript/1302_DeepestLeavesSum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var deepestLeavesSum = function(root) {
    sum = 0;
    depth = 0;
    queue = [];
    
    queue.push({ node: root, height: 0 });
    
    while (queue.length > 0) {
        let { node, height } = queue.shift();
        if ( node === null ) continue;
        
        if (height > depth) {
            deepest = []
            sum = 0;
            depth = height;
        }            
        
        sum += node.val;
        queue.push({ node: node.left, height: height + 1});
        queue.push({ node: node.right, height: height + 1});
    }
    
    return sum;
};
